gabbi/httpclient.py

```diff
@@ -17,12 +17,32 @@
 import os
 import sys
 
-import httplib2
+import urllib3
 
 from gabbi import utils
 
 
-class VerboseHttp(httplib2.Http):
+class Http(urllib3.PoolManager):
+
+    def request(self, absolute_uri, method, body, headers, redirect):
+
+        retries = 1 if redirect else False
```

Comment:
The ideal end goal I'm after is:

* if there is a 302, 303, etc and redirect is True, follow it, otherwise return that status code, don't follow
* in all cases, if there is any error, don't do a retry and force the "real" error to be raised

Is that what your example does. If not, how do I do that in a good clean way?